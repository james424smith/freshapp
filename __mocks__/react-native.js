import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
