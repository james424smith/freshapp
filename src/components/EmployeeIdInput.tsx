import { useTheme } from "@react-navigation/native";
import React from "react";
import InputElement from "./Input";
import Colors from "../constants/styles/colors";
import styles from "../styles/AuthStyles";

interface Props {
  employeeId: string;
  setEmployeeId: (arg: string) => void;
}

const EmployeeIdInput = ({ employeeId, setEmployeeId }: Props) => {
  const theme = useTheme();
  return (
    <InputElement
      testID={"employee-input"}
      containerStyle={{
        ...styles.inputBoxStyle,
        backgroundColor: theme.colors.colorInputText,
      }}
      onChangeText={(newEmployeeId: string) => setEmployeeId(newEmployeeId)}
      value={employeeId}
      inputStyle={{
        ...styles.inputValueStyle,
        color: Colors.white,
      }}
      keyboardAppearance="light"
      placeholder="Employee ID"
      autoFocus={false}
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="numeric"
      returnKeyType="default"
      blurOnSubmit={false}
      placeholderTextColor={Colors.FAQInactiveColor}
      inputContainerStyle={styles.removeUnderline}
      shouldUseFocusedStyles={true}
    />
  );
};
export default EmployeeIdInput;
