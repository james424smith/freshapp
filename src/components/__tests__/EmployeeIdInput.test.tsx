import React from "react";
import EmployeeIdInput from "../EmployeeIdInput";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

describe("EmployeeIdInput snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui and change value of employeeId", () => {
    const setEmployeeId = jest.fn();
    const { toJSON, getByTestId } = render(
      <EmployeeIdInput employeeId={"1234"} setEmployeeId={setEmployeeId} />
    );

    fireEvent.changeText(getByTestId("employee-input"), "12345");
    expect(setEmployeeId).toHaveBeenCalledWith("12345");
    expect(toJSON()).toMatchSnapshot();
  });
});
