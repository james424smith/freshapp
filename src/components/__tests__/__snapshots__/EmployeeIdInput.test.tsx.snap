// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeIdInput snapshot test should render correct ui and change value of employeeId 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 30,
      "height": 86.5,
      "marginBottom": 22.5,
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "zIndex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      autoFocus={false}
      blurOnSubmit={false}
      editable={true}
      keyboardAppearance="light"
      keyboardType="numeric"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Employee ID"
      placeholderTextColor="#8FAABF"
      rejectResponderTermination={true}
      returnKeyType="default"
      secureTextEntry={false}
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFF",
          "flex": 1,
          "flexDirection": undefined,
          "fontFamily": "Roboto-Regular",
          "fontSize": 29.5,
          "marginLeft": 22.5,
          "minHeight": 40,
          "width": 600,
        }
      }
      testID="employee-input"
      underlineColorAndroid="transparent"
      value="1234"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
